### PR TITLE
refactor(linter)!: overhaul config and options

### LINKED ISSUE #5046: feat(linter): Support `plugins` array in config
Support `config.plugins` in `.oxlintrc.json`.

```jsonc
{
  "plugins": ["import", "react-hooks", "@typescript-eslint"],
  // ...
}
```

## Acceptance Criteria
- Should match behavior of eslint <= v8
- Listed plugins should be enabled. We don't need to support disabling plugins this way
- `typescript-eslint` should be handled flexibly. Both `@typescript-eslint` and `typescript-eslint` should be acceptable values.
- Derived JSON schema should provide suggestions for available plugins

### PR DESCRIPTION
**Pull request by DonIsaac**
> Apologies for how large this is, I couldn't split it up any further...

# What This PR Does

Removes `OxlintConfig` and `OxlintOptions` in favor of `LinterBuilder` and `Oxlintrc`. Also adds support for plugins in config files. This should be the last preparation PR needed before I can finally tackle monorepo configs.

## Tl;Dr
- `OxlintConfig` has been replaced with `Oxlintrc`. This gets transformed by `LinterBuilder` into internal `LinterConfig` and `LinterOptions` objects, which the rules and `LintContext` consume.
- Replaces `OxlintOptions` with `LinterBuilder`. Most fields in options are now in `Oxlintrc`. Those that aren't (e.g. --fix) are settable via builder methods.
- Removes `LintPluginOptions` in favor of `LintPlugins`. This makes the API much more succinct.
- Changes how CLI flags are merged with `.oxlintrc.json` files. The process is now:
  1. Read in the config file provided to the CLI to create a `LinterBuilder`. A default one is made when there's no file.
  2. Apply filters/plugins/etc set via CLI args _on top_ of what's initially set from the config file.

### CODE REVIEW

**Comment by DonIsaac** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5333?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#5333** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5333?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#5331** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5331?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5329** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5329?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5326** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5326?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5142** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5142?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5141** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5141?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5140** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5140?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5139** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5139?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5102** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5102?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5101** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5101?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5100** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5100?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @DonIsaac and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>
